atomisticparsers/h5md/parser.py

```diff
@@ -0,0 +1,784 @@
+                if value is not None:
+                    self._system_info[sec_key][key] = value
+                else:
+                    continue
+
+                if isinstance(self._system_info[sec_key][key], h5py.Group):
+                    self._system_info[sec_key][key] = self._data_parser.get_value(self._system_info[sec_key][key], 'value')
+                    if self._system_info[sec_key][key] is None:
+                        continue
+                    elif len(self._system_info[sec_key][key]) != n_frames:  # TODO Should really check that the stored times for these quantities are exact, not just the same length
+                        self.logger.warning('Distinct trajectory lengths of particle attributes not supported. These attributes will not be stored.')
+                        continue
+                else:
+                    self._system_info[sec_key][key] = [self._system_info[sec_key][key]] * n_frames
+
+            # TODO Should we extend this to pick up additional attributes in the particles group? Or require that we follow the H5MD schema strictly?
+
+            # get the system box quantities
+            box = self._data_parser.get_value(particles_group, 'box')
+            if box is None:
+                return
+
+            box_attributes = {'dimension': 'system', 'periodic': 'system'}
+            for box_key, sec_key in box_attributes.items():
+                value = self._data_parser.get_attribute(box, self._nomad_to_box_group_map[box_key], path=None)
+                if value is not None:
+                    self._system_info[sec_key][box_key] = [value] * n_frames
+
+            box_keys = {'lattice_vectors': 'system'}
+            for box_key, sec_key in box_keys.items():
+                value = self._data_parser.get_value(box, self._nomad_to_box_group_map[box_key])
+                if value is not None:
+                    self._system_info[sec_key][box_key] = value
+                else:
+                    continue
+
+                if isinstance(self._system_info[sec_key][box_key], h5py.Group):
+                    self._system_info[sec_key][box_key] = self._data_parser.get_value(self._system_info[sec_key][box_key], 'value')
+                    if self._system_info[sec_key][box_key] is None:
+                        continue
+                    elif len(self._system_info[sec_key][box_key]) != n_frames:  # TODO Should really check that the stored times for these quantities are exact, not just the same length
+                        self.logger.warning('Distinct trajectory lengths of box vectors and positions is not supported. These values will not be stored.')
+                        continue
+                else:
+                    self._system_info[sec_key][box_key] = [self._system_info[sec_key][box_key]] * n_frames
+        return self._system_info
+
+    @property
+    def observable_info(self):
```

Comment:
looking at observable_info, I think we do not need to make this a property. you can simply have a function
```python
def get_observable_info(self, observable_type)

Reply:
yeah, it's not super clear to me when I should be using @property, I need to review this. Does this apply to the other `xxx_info()` functions as well?

Reply:
I will try to look at the others also. but the rule is, we use property if a variable is dependent on some state and needs  to be updated. in this case, it is constant and we should simply use a private variable or a funtion to calc it.

Reply:
that makes sense, thanks. I think probably none of these `xxx_info()` need to be properties then, then are simply parsing the hdf5 groups.

Reply:
converted all these `xxx_info()` to parsing functions now (no longer properties).